### PR TITLE
[UI] fix tab widget on mac

### PR DESCRIPTION
**Pull request by 3nids**
image

superseds #35230

### CODE REVIEW

File: src/gui/qgsproxystyle.h [hunk 1]
```diff
@@ -57,6 +57,8 @@ class GUI_EXPORT QgsAppStyle : public QProxyStyle
     explicit QgsAppStyle( const QString &base );
     QPixmap generatedIconPixmap( QIcon::Mode iconMode, const QPixmap &pixmap, const QStyleOption *opt ) const override;
 
+    QString baseStyle() const {return mBaseStyle;}
```
Comment (nirvn) [inline src/gui/qgsproxystyle.h]:
```suggestion
    QString baseStyle() const { return mBaseStyle; }
```

File: src/app/qgisappstylesheet.cpp [hunk 2]
```diff
@@ -106,18 +108,24 @@ void QgisAppStyleSheet::buildStyleSheet( const QMap<QString, QVariant> &opts )
...
   // Specifically setting a custom close button temporarily works around issue.
-  // TODO: Remove when regression is fixed (Qt 5.9.3 or 5.10?); though hard to tell,
-  //       since we are overriding the default close button image now.
   if ( mMacStyle )
   {
     ss += QLatin1String( "QTabBar::close-button{ image: url(:/images/themes/default/mIconCloseTab.svg); }" );
     ss += QLatin1String( "QTabBar::close-button:hover{ image: url(:/images/themes/default/mIconCloseTabHover.svg); }" );
   }
+#endif
+  if ( mMacStyle )
```
Comment (nirvn) [inline src/app/qgisappstylesheet.cpp]: @3nids , have you tried non-default themes (night mapping and/or blend of gray) to make sure this unconditional stylesheet addition doesn't mess those up. 99.9% it won't, but I don't have a mac at-hand to confirm.

**Review by nirvn — APPROVED** [review]
Much better approach, glad it works out fine.

**Comment by nirvn** [comment]
@3nids , BTW, would be good to backport this to LTR (and 3.12).

File: src/app/qgisappstylesheet.cpp (hunk 2 above)
Comment (3nids) [reply src/app/qgisappstylesheet.cpp]: yes it does look good

**Pull request merged**